### ISSUE TITLE
Unable to pull images: Client.Timeout exceeded while awaiting headers

### ISSUE THREAD

**Issue by deeptuuk** [kubernetes/minikube#3898]
Command:
    minikube start
Output:
😄  minikube v0.35.0 on linux (amd64)
🔥  Creating virtualbox VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
💿  Downloading Minikube ISO ...
 184.42 MB / 184.42 MB [============================================] 100.00% 0s
📶  "minikube" IP address is 192.168.99.100
🐳  Configuring Docker as the container runtime ...
✨  Preparing Kubernetes environment ...
💾  Downloading kubeadm v1.13.4
💾  Downloading kubelet v1.13.4
🚜  Pulling images required by Kubernetes v1.13.4 ...
❌  Unable to pull images, which may be OK: running cmd: sudo kubeadm config images pull --config /var/lib/kubeadm.yaml: command failed: sudo kubeadm config images pull --config /var/lib/kubeadm.yaml
stdout: 
stderr: failed to pull image "k8s.gcr.io/kube-apiserver:v1.13.4": output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
: Process exited with status 1
🚀  Launching Kubernetes v1.13.4 using kubeadm ... 
💣  Error starting cluster: kubeadm init: 
sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI 

[init] Using Kubernetes version: v1.13.4
[preflight] Running pre-flight checks
	[WARNING Service-Docker]: docker service is not enabled, please run 'systemctl enable docker.service'
	[WARNING Swap]: running with swap on is not supported. Please disable swap
[preflight] Pulling images required for setting up a Kubernetes cluster
[preflight] This might take a minute or two, depending on the speed of your internet connection
[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
error execution phase preflight: [preflight] Some fatal errors occurred:
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-apiserver:v1.13.4: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-controller-manager:v1.13.4: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-scheduler:v1.13.4: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-proxy:v1.13.4: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/pause:3.1: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/etcd:3.2.24: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/coredns:1.2.6: output: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
, error: exit status 1
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`

: Process exited with status 1

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new

i am really frustrated
can you help me,i am really appreciate you

**Comment by laozc** [comment]
It looks like you cannot access k8s.gcr.io.
If you're a user from Chinese mainland, you may specify a mirror from Aliyun once #3714 is merged.

**Comment by alexlatchford** [comment]
I'm getting the same error at the moment and I'm in Seattle :/

**Comment by laozc** [comment]
It looks like your case is different.
The log indicates that k8s.gcr.io was unable to resolve with your DNS server 192.168.64.1:53.

	[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-apiserver:v1.13.4: output: Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on 192.168.64.1:53: read udp 192.168.64.5:40959->192.168.64.1:53: read: connection refused
	[ERROR ImagePull]: failed to pull image k8s.gcr.io/kube-controller-manager:v1.13.4: output: Error response from daemon: Get https://k8s.gcr.io/v2/: dial tcp: lookup k8s.gcr.io on 192.168.64.1:53: read udp 192.168.64.5:33827->192.168.64.1:53: read: connection refused

You may check if your networking configuration is correct inside the VM.

**Comment by deeptuuk** [comment]
> It looks like you cannot access k8s.gcr.io.
> If you're a user from Chinese mainland, you may specify a mirror from Aliyun once #3714 is merged.

thank you
but for now, how can i let it run?

**Comment by laozc** [comment]
A separate binary is generated for each PR.
You may refer to https://github.com/kubernetes/minikube/blob/master/docs/contributors/ci_builds.md for more details on how to download the binaries.
e.g. https://storage.googleapis.com/minikube-builds/3714/minikube-linux-amd64

**Comment by laozc** [comment]
PR #3714 is merged.
You'll be seeing this change in the next release with this flag `--image-repository registry.cn-hangzhou.aliyuncs.com/google_containers`.

There is another PR #3937 under review which may help you choose an accessible image mirror if you live in Chinese mainland. With this change you should be able to run minikube with a local mirror.

**Comment by tstromberg** [comment]
This error happens due to a bad internet or HTTP proxy configuration. We now point users to the HTTP proxy documentation when this error occurs: 

https://github.com/kubernetes/minikube/blob/master/docs/http_proxy.md

Thanks for filing this bug report!

**Comment by walls00000** [comment]
Can someone update the link?

**Comment by cpaika** [comment]
@walls00000 https://minikube.sigs.k8s.io/docs/handbook/vpn_and_proxy/

**Comment by walls00000** [comment]
Thanks @cpaika.  Documentation looks good.
